tsl/src/compression/compression.c

```diff
@@ -441,6 +443,12 @@ compress_chunk_sort_relation(Relation in_rel, int n_keys, const ColumnCompressio
 
 	heap_endscan(heapScan);
 
+	/* Perform an analyze on the chunk to get up-to-date stats before compressing.
+	 * We do it at this point to reduce the amount of random IO operations
+	 * necessary in case the statistic target is larger than the chunk heap size.
```

Comment:
We do it here because we just read out the entire chunk into tuplesort, its pages are cached and we can save on disk I/O, right? The wording with random I/O and statistics target is somewhat confusing...

Reply:
That's true. I'm open to suggestions about better wording :smile:

Reply:
"We do it at this point because we've just read out the entire chunk into tuplesort, so its pages are likely to be cached and we can save on I/O".

Although I wonder to which extent this is true, because full heap scans use a dedicated cache strategy which doesn't cache much.

Reply:
I have definitely seen a rise in average read op size. In any case, heap will be present in os/disk cache if not shared buffers.

Reply:
The Postgres cache uses the dedicated strategy but the os/disk cache doesn't and that's the caching we really care about.

Reply:
Updated comment to suggestion.